pkg/app/app.go

```diff
@@ -1398,6 +1411,35 @@ Do you really want to apply?
 	return true, true, applyErrs
 }
 
+func getReleasesWithPreApply(releases []state.ReleaseSpec) []state.ReleaseSpec {
```

Comment:
The functions of this function and https://github.com/helmfile/helmfile/blob/main/pkg/event/bus.go#L54 Repeat, is it necessary?  @Sajfer

Reply:
I dont understand what you mean, they are not the same?

Reply:
bus.go will check if the presync exist.

Reply:
@Sajfer The `Trigger` function does execute the hook only when the hook is defined for the said event(in this case `preapply`), and therefore you don't need to filter releases(along with their hooks) beforehand with `slices.Contains(hook.Events, "preapply")`.

Also, the list of releases passed to this function (which is `toApplyWithNeeds`) contains `releasesWithNoChange` which are basically unaffected releases. We shouldn't call preapply hooks for unaffected releases, right? Then we shouldn't include those.

Reply:
@Sajfer Would you mind confirming https://github.com/helmfile/helmfile/pull/79/files#r973672208 ?